### PR TITLE
Oneapi lapack

### PR DESCRIPTION
**Pull request by syurkevi**
This PR adds several missing lapack functions - svd, cholesky, lu to the oneapi backend, as well as the identity kernel required for the svd tests.

Description
-----------
* A few tests are still reliant on the indexing kernel and it is unknown if they are passing. 
* Adds lu, with kernel::lu_split
* Adds svd (inplace, out-of-place)
* Adds cholesky
Tests also rely on blas functions in outstanding blas PR.

Checklist
---------

- [x] Rebased on latest master
- [x] Code compiles
- [x] Tests pass

### CODE REVIEW

File: vcpkg.json [hunk 1]
```diff
@@ -23,10 +23,7 @@
     ],
     "features": {
         "tests": {
-            "description": "Build with tests",
-            "dependencies": [
-                "gtest"
-            ]
+            "description": "Build with tests"
```
Comment (umar456) [inline vcpkg.json:26]: I don't remember why we did this. Can you add this back?

File: test/iota.cpp [hunk 2]
```diff
@@ -102,6 +102,8 @@ using af::array;
 using af::iota;
 
 TEST(Iota, CPP) {
+    af::setDevice(0);
```
Comment (umar456) [inline test/iota.cpp:105]: I think this was an unintended change. please remove.

File: test/blas.cpp [hunk 3]
```diff
@@ -119,6 +127,11 @@ TYPED_TEST(MatrixMultiply, NonSquare) {
 }
 
 TYPED_TEST(MatrixMultiply, SquareVector) {
+    af::setDevice(0);
+    af::info();
+    af::array a = af::randu(3, 3);
+    a.eval();
+    af_print(a);
```
Comment (umar456) [inline test/blas.cpp:134]: Debugging changes should be removed from tests

File: src/backend/oneapi/CMakeLists.txt [hunk 4]
```diff
@@ -264,15 +266,20 @@ add_library(ArrayFire::afoneapi ALIAS afoneapi)
 
 arrayfire_set_default_cxx_flags(afoneapi)
 
+find_package(MKL CONFIG REQUIRED)
+message(STATUS "${MKL_IMPORTED_TARGETS}")
```
Comment (umar456) [inline src/backend/oneapi/CMakeLists.txt:270]: Master now has the changes to correctly include the MKLConfig file from the oneapi install. Please remove

**Review by umar456 — CHANGES_REQUESTED** [review]
Looks like you are still working on this PR. I left a few comments but I think you know about most of them.

File: vcpkg.json (hunk 1 above)
Comment (syurkevi) [reply vcpkg.json:26]: gtest binaries are incompatible w/IntelLLVM and cause SEH exceptions. we will need a separate toolchain file. to be addressed in separate PR

File: src/backend/oneapi/platform.cpp [hunk 5]
```diff
@@ -172,7 +172,8 @@ string getDeviceInfo() noexcept {
                 device->get_info<sycl::info::device::driver_version>();
             info << devVersion;
             info << " -- Device driver " << driVersion;
-
+            info << " -- FP64 Support: "
```
Comment (umar456) [inline src/backend/oneapi/platform.cpp:175]: Can you remove this change?

File: vcpkg.json (hunk 1 above)
Comment (umar456) [reply vcpkg.json:26]: Can you not add this to the main vcpkg and only do it locally while we figure out a solution. This change will break the CI and all the other backends.

File: src/backend/oneapi/kernel/iota.hpp [hunk 6]
```diff
@@ -95,7 +95,7 @@ void iota(Param<T> out, const af::dim4& sdims) {
 
     try {
         getQueue()
-            .submit([=](sycl::handler& h) {
+            .submit([&](sycl::handler& h) {
```
Comment (umar456) [inline src/backend/oneapi/kernel/iota.hpp:98]: we want this by value here right? otherwise the objects will be deleted when they go out of scope.

**Review by umar456 — APPROVED** [review]

**Review by umar456 — DISMISSED** [review]

**Review by umar456 — APPROVED** [review]

**Pull request merged**